gumby/sync.py

```diff
@@ -419,10 +476,99 @@ def proto_go(self, line):
         if line.strip().startswith("go:"):
             start_delay = max(0, float(line.strip().split(":")[1]) - time())
             self._logger.info("Starting the experiment in %f secs.", start_delay)
-            reactor.callLater(start_delay, self.startExperiment)
+            reactor.callLater(start_delay, self.start_experiment)
             self.factory.stopTrying()
             self.transport.loseConnection()
 
+    def register_callbacks(self, module):
+        member_names = [name for name in dir(module) if type(getattr(module.__class__, name, None)).__name__ != "property"]
+        for member in [getattr(module, key) for key in member_names]:
+            if not (callable(member) and hasattr(member, "register_as_callback")):
+                continue
+            else:
+                self.scenario_runner.register(member, name=member.register_as_callback)
+
+    @experiment_callback
+    def echo(self, *argv):
+        self._logger.info("%s ECHO %s", self.my_id, ' '.join(argv))
+
+    @experiment_callback
+    def annotate(self, message):
+        self._stats_file.write('%.1f %s %s %s\n' % (time(), self.my_id, "annotate", message))
+
+    @experiment_callback
+    def peertype(self, peer_type):
+        self._stats_file.write('%.1f %s %s %s\n' % (time(), self.my_id, "peertype", peer_type))
+
+    @experiment_callback
+    def stop(self):
+        stop_reactor()
+
+    def print_on_change(self, name, prev_dict, cur_dict):
```

Comment:
This is a relic. It came from the old dispersy experiment client. Some of the older scenarios use it. They keep a dict, and periodically (usually a LoopingCall), construct a new dict and pass the old and new in. Then some changed variables would get printed and the new one stored as old for the next cycle. I dunno how usefull or not it is.

As to the naming, you're right, we might as well change it. The `_on_` term suggests an event or something automatic, which this is not. So perhaps `print_dict_changes`?